lib/data_services_api/dataset.rb

```diff
@@ -49,15 +49,54 @@ def structure
     end
 
     def query(query)
-      service.api_post_json(data_api, query.to_json)
+      converter = SapiNTConverter.new(query.to_json)
+      sapi_query_params = converter.to_sapint_query
+      sapint_response = service.api_get_json(data_api, sapi_query_params)
+      to_dsapi_response(sapint_response)
     end
 
     def describe(uri)
       service.api_get_json(describe_api, uri: uri)
     end
 
     def explain(query)
-      service.api_post_json(explain_api, query.to_json)
+      converter = SapiNTConverter.new(query.to_json)
+      sapi_query_params = converter.to_sapint_query
+      explain_url = "#{data_api}/explain"
+      { sparql: service.api_get_text(explain_url, sapi_query_params) }
+    end
+
+    private
+
+    # Converts SAPINT returned JSON format to DSAPI returned JSON format
+    def to_dsapi_response(sapint_response)
+      sapint_response['items'].map do |value|
+        to_dsapi_item(value)
+      end
+    end
+
+    def to_dsapi_item(sapint_item)
+      dsapi_item = {}
```

Comment:
[map](https://ruby-doc.org/core-3.0.1/Enumerable.html#method-i-map) is defined on `Enumerable`, so can be used directly on  a `Hash`:

```ruby
def to_dsapi_item(sapint_item)
  sapint_item.map { |k, v| [k, to_dsapi_json(k, v)] }.to_h
end
```